python/ray/autoscaler/aws/example-cloudwatch.yaml

```diff
@@ -0,0 +1,116 @@
+            # Note: To allow per-node alarm being created and updated, we recommend "{instance_id}" included as part of "AlarmName" in the following json config.
+            config: "cloudwatch/example-cloudwatch-alarm-config.json"
+
+# How Ray will authenticate with newly launched nodes.
+auth:
+    ssh_user: ubuntu
+
+available_node_types:
+  ray.head.default:
+      node_config:
+        InstanceType: c5a.large
+        ImageId: latest_dlami
+      resources: {}
+      max_workers: 10
+  ray.worker.default:
+      node_config:
+        InstanceType: c5a.large
+        ImageId: latest_dlami
+        # Note: IamInstanceProfile is needed to grant worker nodes required permission to make boto3 call for Cloudwatch setup.
+        # Default IamInstanceProfile is arn:aws:iam::{your_aws_account_number}:instance-profile/ray-autoscaler-cloudwatch-v1.
+        IamInstanceProfile:
+          Arn: arn:aws:iam::{your_aws_account_ID}:instance-profile/ray-autoscaler-cloudwatch-v1
+      resources: {}
+      min_workers: 1
+head_node_type: ray.head.default
+
+# Files or directories to copy to the head and worker nodes. The format is a
+# dictionary from REMOTE_PATH: LOCAL_PATH, e.g.
+# If you want to export Ray system-level Prometheus metrics to Cloudwatch,
+# Uncomment below commands under file_mounts and head_setup_commands.
+# Commands under these two sections all serve for exporting Prometheus metrics to
+# CloudWatch. See https://docs.ray.io/en/master/ray-metrics.html for Ray prometheus support.
+#file_mounts: {
+#    "~/cloudwatch/prometheus.yml": "cloudwatch/prometheus.yml",
+#    "~/cloudwatch/ray_prometheus_waiter.sh": "cloudwatch/ray_prometheus_waiter.sh"
+#}
+
+# Custom commands that will be run on the head node after common setup.
+# Uncomment below if you want to export Ray system-level Prometheus metrics to Cloudwatch.
+# Note: Prometheus to Cloudwatch support requires cloudwatch agent installed
+#head_setup_commands:
+#    - sudo cp -f /home/ubuntu/cloudwatch/prometheus.yml /opt/aws/amazon-cloudwatch-agent/etc
+#    - sudo cp -f /home/ubuntu/cloudwatch/ray_prometheus_waiter.sh /opt/aws/amazon-cloudwatch-agent/bin
+#    - sudo chmod +x /opt/aws/amazon-cloudwatch-agent/bin/ray_prometheus_waiter.sh
+#    - nohup sudo sh -c "/opt/aws/amazon-cloudwatch-agent/bin/ray_prometheus_waiter.sh `cat ~/ray_bootstrap_config.yaml | python -c 'import sys, json; print(json.load(sys.stdin)["cluster_name"])' ` >> '/opt/aws/amazon-cloudwatch-agent/logs/ray_prometheus_waiter.out' 2>> '/opt/aws/amazon-cloudwatch-agent/logs/ray_prometheus_waiter.err'" &
+
+# Since CloudWatch Unified Agent install can take 15-60 minutes on a newly launched
+# Ubuntu deep-learning AMI, it's not always practical to install it on every node of
+# a large autoscaling cluster. As a best practice, you should launch a single head node
+# with CloudWatch Agent installed, create an AMI from that instance, and then launch a
+# new Ray cluster from that AMI.
+#setup_commands:
```

Comment:
I'd recommend separating this out into two example YAMLs - one that illustrates how to create an Ubuntu DL AMI with CloudWatch pre-installed, and another that illustrates how to then either use this AMI (or an existing, public Amazon Linux AMI?) to eliminate CloudWatch Agent installation latency overhead.